apps/prairielearn/src/lib/groups.sql

```diff
@@ -142,10 +141,11 @@ SELECT
   bool_or(gr.can_assign_roles_during_assessment) AS can_assign_roles_during_assessment
 FROM
   group_roles as gr
-  JOIN group_user_roles as gu ON gr.id = gu.group_role_id
+  JOIN group_user_roles as gur ON (gr.id = gur.group_role_id)
 WHERE
   gr.assessment_id = $assessment_id
-  AND gu.user_id = $user_id;
+  -- TODO probably a good idea to add a join with groups before doing this check?
+  AND gur.user_id = $user_id;
```

Comment:
https://github.com/PrairieLearn/PrairieLearn/pull/9112#discussion_r1437271630

Reply:
I agree that we should filter down this table based on `group_id` to avoid a table scan.

Reply:
Done in 175015e43. @nwalters512 can you double check to make sure I didn't mess something up? I removed the use of the assessment_id column from group_roles, since I'm now filtering on group_configs, I believe the result should be the same.

Reply:
I believe that looks good to me.